src/App.js

```diff
@@ -1,26 +1,114 @@
+  InputGroup,
+  InputGroupAddon,
+  InputGroupText,
+  Col,
+  Row,
+  Label,
+  Input
+} from 'reactstrap';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  state = {
+    income: 0,
+    buckets: {
+      "Fixed Costs": 50,
+      Investments: 10,
+      Savings: 20,
+      "Guilt-Free Spending": 20,
+    }
+  }
+
+  updateIncome = e => {
+    this.setState({ income: e.target.value });
+  };
+
+  percentUpdated = (bucketName, percentage) => {
+    let buckets = this.state.buckets;
+    buckets[bucketName] = percentage
+    this.setState(buckets)
+  }
+
+  renderBuckets = () => {
+    const { buckets } = this.state
+
+    return Object.keys(buckets).map((bucketName, i) => {
+      return (
+        <Bucket
+          key={i}
+          name={bucketName}
+          percentage={buckets[bucketName]}
+          percentUpdated={this.percentUpdated}
+          income={this.state.income}
+        />
+      )
+    });
+  };
+
+  renderTotal = () => {
+    const total = Object.values(this.state.buckets).reduce(
+      (sum, currentValue) => Number(sum) + Number(currentValue)
+    );
+
+    return (
+      <Row className="mt-2">
+        <Col md="2">
+          <Label>Total</Label>
+        </Col>
+        <Col md={{size: 5, offset: 1}}>
+          <InputGroup>
+            <Input
+              className="form-control"
+              type="number"
+              id="total"
+              value={total}
+              readOnly
+            />
+            <InputGroupAddon addonType="append">
+              <InputGroupText>%</InputGroupText>
+            </InputGroupAddon>
+          </InputGroup>
+        </Col>
+      </Row>
+    )
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <h2>IWT Budget Calculator</h2>
+        <Container>
+          <Row>
+            <Col md="2">
+              <Label>Income</Label>
+            </Col>
+            <Col md={{size: 5, offset: 1}}>
+              <InputGroup>
+                <InputGroupAddon addonType="prepend">
+                  <InputGroupText>$</InputGroupText>
+                </InputGroupAddon>
+                <Input
+                  className="form-control"
+                  type="number"
+                  defaultValue={0}
+                  onChange={this.updateIncome}
+                />
+              </InputGroup>
+            </Col>
+            <Col md="2">
+              <span>{this.state.income}</span>
+            </Col>
+          </Row>
+          { this.renderBuckets() }
+          { this.renderTotal() }
```

Comment:
Maybe since you're rendering two functions like this it could be an opportunity to break them down into different files